src/main/java/com/google/devtools/build/lib/analysis/actions/StrippingPathMapper.java

```diff
@@ -145,7 +145,9 @@ static Optional<PathMapper> tryCreate(AbstractAction action, boolean isStarlarkA
 
   @Override
   public String getMappedExecPathString(ActionInput artifact) {
-    if (isSupportedInputType(artifact) && isOutputPath(artifact, outputRoot)) {
+    if (artifact instanceof DerivedArtifact
```

Comment:
I found this easier to read with the helper method. Perhaps keep an `isSupported(artifact)` that does both the type check and calls `isOutputPath`.